ontimize/components/form/o-form.component.ts

```diff
@@ -788,10 +816,22 @@ export class OFormComponent implements OnInit, OnDestroy, CanComponentDeactivate
       self.postCorrectInsert(resp);
       self.formCache.setCacheSnapshot();
       self.markFormLayoutManagerToUpdate();
-      if (self.stayInRecordAfterInsert) {
+      if (self.stayInRecordAfterInsert || self.afterInsertMode === 'detail') {
         self._stayInRecordAfterInsert(resp);
+        if (self.afterInsertMode === 'new') {
+          let form = <HTMLFormElement>this.elRef.nativeElement.getElementsByTagName('form')[0];
+          form.reset();
```

Comment:
Check `setFormMode` method due to this behaviour shold be similar to turn to insert mode